rpnpy/calculator.py

```diff
@@ -536,28 +538,30 @@ def _tryEvalExec(self, command, modifiers, count):
                 possibleWhiteSpace = True
 
             try:
-                exec(command, globals(), self._variables)
-            except BaseException as e:
-                err = str(e)
-                errors.append('Could not exec(%r): %s' % (command, err))
-                if (not possibleWhiteSpace and self._splitLines and
-                        err.startswith(
-                            'unexpected EOF while parsing (<string>, '
-                            'line 1)')):
-                    possibleWhiteSpace = True
-
-                if possibleWhiteSpace:
-                    errors.append('Did you accidentally include whitespace '
-                                  'in a command line?')
-                raise CalculatorError(*errors)
-            else:
-                self.debug('exec(%r) worked.' % command)
-                return True, self.NO_VALUE
-        else:
-            self.debug('eval %s worked: %r' % (command, value))
-            count = 1 if count is None else count
-            self._finalize(value, modifiers=modifiers, repeat=count)
-            return True, value
+                value = EngNumber(command)
+            except decimal.InvalidOperation:
+                try:
+                    exec(command, globals(), self._variables)
+                except BaseException as e:
+                    err = str(e)
+                    errors.append('Could not exec(%r): %s' % (command, err))
+                    if (not possibleWhiteSpace and self._splitLines and
+                            err.startswith(
+                                'unexpected EOF while parsing (<string>, '
+                                'line 1)')):
+                        possibleWhiteSpace = True
+
+                    if possibleWhiteSpace:
+                        errors.append('Did you accidentally include whitespace '
+                                      'in a command line?')
+                    raise CalculatorError(*errors)
+                else:
+                    self.debug('exec(%r) worked.' % command)
+                    return True, self.NO_VALUE
+        self.debug('eval %s worked: %r' % (command, value))
```

Comment:
Shouldn't there be an `else` here at the same level as the `try/except` on the `eval`. It seems to be gone from the diff and this code block doesn't look indented enough.

Reply:
The issue is that that `else` block not only needs to be executed when the `eval`'s `try` suceeds, but also when it fails and the `EngNumber`'s `try` suceeds. So I removed the `else` so that it can also execute on the `EngNumber` suceeding path.

That doesn't cause side effects because the other two paths inside the `except decimal.InvalidOperation` either `raise`s or `return`s.

Another option would be to put the `else` back, but duplicate that code to inside an `else` on the `EngNumber`'s `try`. Do you think that would be better?

Reply:
I think duplicating it would be good. The reason I say that is to keep the debug message accurate, to say what exactly it is that worked.

I hope I'm not saying things that are inaccurate!  I'm just adding these comments based on looking at snippets of the diff. It's great you're doing this, and I hope I'm not causing too much friction or getting you impatient. I'm happy to merge this immediately if you think it's fine to add in the extra/specific else. Or you can just tell me to merge it :-)

I'm working maybe 100 hours a week on other stuff....... but I use `rpn.py` every single day, so am very happy to see it moving on & that you're into it and sending along pull requests. Thanks!

Reply:
> I think duplicating it would be good. The reason I say that is to keep the debug message accurate, to say what exactly it is that worked.

Makes total sense!

> I hope I'm not saying things that are inaccurate! I'm just adding these comments based on looking at snippets of the diff. It's great you're doing this, and I hope I'm not causing too much friction or getting you impatient. I'm happy to merge this immediately if you think it's fine to add in the extra/specific else. Or you can just tell me to merge it :-)

Not at all! I'm all for taking the time to do it properly the first time around. No rush from my side. In the meantime I'm running rpnpy using these commits so I have access to the engineering notation anyway :).

> I'm working maybe 100 hours a week on other stuff....... but I use rpn.py every single day, so am very happy to see it moving on & that you're into it and sending along pull requests. Thanks!

Glad to hear that you still use it daily, so I can count on feedback from you if this feels off ;).

Oh, and I just pushed the duplication of the else block, so there's a debug message specific to rpnpy, and everything works fine. Let me know if there are any improvements left to be done!